src/app/shared/types/modal.ts

```diff
@@ -6,6 +6,7 @@ export interface ModalData {
   fileExt?: string;
   fileTypes?: string[];
   fileNum?: number;
+  documentSource?: string;
```

Comment:
Creating an ENUM could make things stricter. Just a suggestion, not a requirement.

Reply:
Good suggestion! This is a good case for an enum actually